pkg/virtual/initializingworkspaces/builder/caching_authorizer.go

```diff
@@ -0,0 +1,71 @@
+package builder
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	kcpkubernetesclientset "github.com/kcp-dev/client-go/kubernetes"
+	"github.com/kcp-dev/kcp/pkg/apis/tenancy/initialization"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/authorization/delegated"
+	dynamiccontext "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/context"
+	"github.com/kcp-dev/logicalcluster/v2"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/klog/v2"
+)
+
+func newCachingAuthorizer(client kcpkubernetesclientset.ClusterInterface) authorizer.Authorizer {
```

Comment:
We're trying to return specific types whenever possible (xref https://github.com/golang/go/wiki/CodeReviewComments#interfaces) so I think you can return a `*cachingAuthorizer` here.